--- gamesummary.inc.orig        2019-01-11 01:07:16.246386411 +0100
+++ gamesummary.inc     2019-01-11 01:09:30.131137088 +0100
@@ -4,6 +4,8 @@
   <dd class="gs_map"><%map.title.safe%></dd>
   <dt class="gs_players">Players</dt>
   <dd class="gs_players"><%rules.numplayers%>/<%rules.maxplayers%></dd>
-  <dt class="gs_time">Time</dt>
-  <dd class="gs_time"><%time.elapsed%>/<%time.timelimit%></dd>
+  <!-- dt class="gs_time">Time</dt>
+  <dd class="gs_time"><%time.elapsed%>/<%time.timelimit%></dd -->
+  <dt class="gs_wave">Wave <%wave.num%></dt>
+  <dd class="gs_wave"><%wave.monsters.dead%>/<%wave.monsters.total%></dd>
 </dl>
\ No newline at end of file
