--- crudini.orig        2022-12-21 13:34:52.501451754 +0000
+++ crudini     2022-12-21 13:35:27.150704760 +0000
@@ -17,15 +17,16 @@
 import hashlib
 import iniparse
 import os
-import pipes
 import shutil
 import string
 import tempfile

 if sys.version_info[0] >= 3:
+    import shlex as pipes
     from io import StringIO
     import configparser
 else:
+    import pipes
     from cStringIO import StringIO
     import ConfigParser as configparser
