diff -ruN ServerAdmin.orig/current_player_row.inc ServerAdmin/current_player_row.inc
--- ServerAdmin.orig/current_player_row.inc	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/current_player_row.inc	2022-12-22 01:43:52.990541726 +0100
@@ -1,7 +1,7 @@
 <tr class="<%evenodd%>">
   <td style="background: <%player.teamcolor%>; color: <%player.teamcolor%>;"><%player.teamid%>&#160;</td>
   <td><%player.name%></td>
-  <td><%player.perk.name%></td>
+  <td class="foo-bar"><%player.perk.name%>;<%player.perk.level%>;<%player.playerkey%></td>
   <td class="right"><%player.score%></td>
   <td class="right"><%player.pawn.health%></td>
   <td class="right"><%player.kills%></td>
diff -ruN ServerAdmin.orig/current_players_row.inc ServerAdmin/current_players_row.inc
--- ServerAdmin.orig/current_players_row.inc	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/current_players_row.inc	2022-12-22 03:03:42.614832917 +0100
@@ -2,9 +2,9 @@
   <td style="background: <%player.teamcolor%>; color: <%player.teamcolor%>;"><%player.teamid%>&#160;</td>
   <td><%player.name%></td>
   <td class="right"><%player.ping%></td>
-  <td><%player.ip%></td>
+  <td><a href="https://geoiplookup.net/ip/<%player.ip%>" target="_blank"><%player.ip%></a></td>
   <td><%player.uniqueid%></td>
-  <td><%player.steamid%></td>
+  <td><a href="https://steamcommunity.com/profiles/<%player.steamid%>" target="_blank"><%player.steamid%></a></td>
   <td><%player.steamname%></td>
   <td class="center"><%player.admin%></td>
   <td class="center"><%player.spectator%></td>
diff -ruN ServerAdmin.orig/current_rules.inc ServerAdmin/current_rules.inc
--- ServerAdmin.orig/current_rules.inc	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/current_rules.inc	2022-12-22 01:43:52.990541726 +0100
@@ -1,3 +1,7 @@
+<dt>Wave</dt>
+<dd><%wave.num%></dd>
+<dt>Difficulty</dt>
+<dd><%rules.difficulty.text%></dd>
 <dt>Players</dt>
 <dd><%rules.numplayers%>/<%rules.maxplayers%></dd>
 <dt>Minimum to Start</dt>
diff -ruN ServerAdmin.orig/default_maplist.html ServerAdmin/default_maplist.html
--- ServerAdmin.orig/default_maplist.html	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/default_maplist.html	2022-12-22 01:43:52.990541726 +0100
@@ -2,6 +2,8 @@
 
 <div id="content">
 
+<p>Note: the first 2 map cycles are generated, so you cannot make changes to them.</p>
+
 <form id="maplistselect" method="post" action="<%page.fulluri%>">
 <fieldset>
 <legend>Map list selection</legend>
@@ -9,7 +11,7 @@
     <option value="-1" class="em">Create new map cycle</option>
     <%maplists%>
     </select>
-    <button type="submit" id="btnselect">edit map cycle</button>
+    <!--button type="submit" id="btnselect">edit map cycle</button-->
 </fieldset>
 </form>
 
@@ -34,8 +36,8 @@
 <%maplist_editor%>
 
 <div id="mlactions">
-<button type="submit" name="action" value="save">save map cycle</button>
-<button type="submit" name="delete" value="doit" <%mldeletable%> >delete</button>
+<button id="mlactions.save" type="submit" name="action" value="save">save map cycle</button>
+<button id="mlactions.delete" type="submit" name="delete" value="doit" <%mldeletable%> >delete</button>
 <button type="submit" name="activate" value="activate" <%mlactive%> >make active</button>
 </div>
 </fieldset>
@@ -53,6 +55,24 @@
      $('#maplistidx').change(function(){
         $('#maplistselect').submit();
     });
+
+    /* we generate 2 fixed cycles, prohibit changes on those as they'll be overriden with the next klf config */
+    document.getElementById("simpleEditBtn").hidden = true;
+    cycleId = "<%maplistidx%>";
+    if (cycleId <= 1)
+    {
+        document.getElementById("mlactions.save").hidden = true;
+        document.getElementById("mlactions.delete").hidden = true;
+        document.getElementById("jsmlctrls").hidden = true;
+        document.getElementById("jsmapcycle").readOnly = true;
+
+        /* remove the delete buttons reactively
+         * ugly, but this way we can avoid touching js files */
+        [].forEach.call(document.querySelectorAll('.deleteButton'), function (el)
+        {
+            el.style.visibility = 'hidden';
+        });
+    }
 });
     
 //]]>
diff -ruN ServerAdmin.orig/gamesummary.inc ServerAdmin/gamesummary.inc
--- ServerAdmin.orig/gamesummary.inc	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/gamesummary.inc	2022-12-22 01:43:52.990541726 +0100
@@ -4,6 +4,8 @@
   <dd class="gs_map"><%map.title.safe%></dd>
   <dt class="gs_players">Players</dt>
   <dd class="gs_players"><%rules.numplayers%>/<%rules.maxplayers%></dd>
-  <dt class="gs_time">Time</dt>
-  <dd class="gs_time"><%time.elapsed%>/<%time.timelimit%></dd>
+  <!-- dt class="gs_time">Time</dt>
+  <dd class="gs_time"><%time.elapsed%>/<%time.timelimit%></dd -->
+  <dt class="gs_wave">Wave <%wave.num%></dt>
+  <dd class="gs_wave"><%wave.monsters.dead%>/<%wave.monsters.total%></dd>
 </dl>
\ No newline at end of file
diff -ruN ServerAdmin.orig/policy_bans.html ServerAdmin/policy_bans.html
--- ServerAdmin.orig/policy_bans.html	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/policy_bans.html	2022-12-22 01:43:52.990541726 +0100
@@ -2,6 +2,8 @@
 
 <div id="content">
 
+<p>List of the currently applied bans. Use the <code>killinuxfloor ban</code> command to add or remove bans. Apply your changes with <code>killinuxfloor config && killinuxfloor restart</code>.</p>
+
 <h2><%page.title%></h2>
 
 <div class="section narrow">
@@ -13,7 +15,7 @@
     <th>Steam ID</th>
     <th title="The Steam community ID. Can be '[unknown]' when the unique ID could not be resolved to the correct Steam ID.">Community ID</th>
     <!--<th>Timestamp</th>-->
-    <th></th>
+    <!--th></th-->
 </tr>
 </thead>
 <tbody>
@@ -21,7 +23,7 @@
 </tbody>
 </table>
 
-<form method="post" action="<%page.fulluri%>+export">
+<!--form method="post" action="<%page.fulluri%>+export">
 <p><button type="submit" title="Export the current bans to a JSON file.">Export</button></p>
 </form>
 </div>
@@ -67,7 +69,7 @@
 </dl>
 </div>
 </fieldset>
-</form>
+</form-->
 
 </div>
 
diff -ruN ServerAdmin.orig/policy_bans_row.inc ServerAdmin/policy_bans_row.inc
--- ServerAdmin.orig/policy_bans_row.inc	2022-12-22 01:43:19.509416046 +0100
+++ ServerAdmin/policy_bans_row.inc	2022-12-22 01:43:52.990541726 +0100
@@ -1,15 +1,15 @@
 <tr>
   <!--<td><%ban.playername%></td>-->
   <td><%ban.uniqueid%></td>
-  <td><%ban.steamid%></td>
+  <td><a href="https://steamcommunity.com/profiles/<%ban.steamid%>" target="_blank"><%ban.steamid%></a></td>
   <td><%ban.steamname%></td>
   <!--<td><%ban.timestamp%></td>-->
-  <td>
+  <!--td>
     <form method="post" action="<%page.fulluri%>">
     <div>
     <input type="hidden" name="banid" value="<%ban.banid%>" />
     <button type="submit" name="action" value="delete">Delete</button>
     </div>
     </form>
-  </td>
+  </td-->
 </tr>
